### PR TITLE
Add support for proto.to_any

### PR DESCRIPTION
**Pull request by michaelkipper**
Re: https://github.com/stripe/skycfg/issues/45

This simple PR adds support for a `proto.to_any()` function, so we can create schema agnostic protobuf messages with Skycfg.

### CODE REVIEW

**Comment by michaelkipper** [comment]
@jmillikin-stripe: What do you think about this PR?

File: internal/go/skycfg/proto_api.go [hunk 1]
```diff
@@ -234,6 +238,33 @@ func fnProtoToJson(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple
...
+	// Disambiguate between golang and gogo encoded proto messages.
+	var any proto.Message
+	if "" != proto.MessageName(msg.msg) {
+		// Returns a golang any.Any type.
+		any, err = ptypes.MarshalAny(msg.msg)
+	} else if "" != gogo_proto.MessageName(msg.msg) {
+		// Returns a gogo types.Any type.
+		any, err = gogo_types.MarshalAny(msg.msg)
+	} else {
+		return nil, fmt.Errorf("%s: could not get message name for %s", "proto.to_any", msg.Type())
```
Comment (dilyevsky) [inline internal/go/skycfg/proto_api.go]:
```suggestion
		return nil, fmt.Errorf("%s: could not get message name for %s", fn.Name(), msg.Type())
```

File: internal/go/skycfg/proto_api.go [hunk 2]
```diff
@@ -234,6 +238,33 @@ func fnProtoToJson(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple
 	return starlark.String(jsonData), nil
 }
 
+// Implementation of the `proto.to_any()` built-in function. Returns a
+// skyProtoMessage with an `Any` proto.Message in it.
+func fnProtoToAny(t *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var msg *skyProtoMessage
+	err := wantSingleProtoMessage("proto.to_any", args, kwargs, &msg)
```
Comment (dilyevsky) [inline internal/go/skycfg/proto_api.go]:
```suggestion
	err := wantSingleProtoMessage(fn.Name(), args, kwargs, &msg)
```

**Comment by idiamond-stripe** [comment]
Hit this today when trying to use the Envoy API in skycfg. The Envoy `Filter` message takes a `google.protobuf.Any`, which isn't currently possible to implement using Skycfg [0]. This typed config is one of the options in the `extensions/` subdirectory, for example `HTTPConnectionManager` [1].

It's not currently possible to work around as it's not possible to marshal Skycfg to a protobuf so you can put it into the `value` block of `google.protobuf.Any`, nor introspect the type name.

[0] https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/listener/listener_components.proto#L29-L43
[1] https://github.com/envoyproxy/data-plane-api/blob/master/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L41

**Comment by jmillikin-stripe** [comment]
Replaced by https://github.com/stripe/skycfg/pull/83

**Pull request closed without merging**